### ISSUE TITLE
setup.py --help-commands lists "Extra commands" in random order

### ISSUE THREAD

**Issue by jtmoon79** [pypa/setuptools#1942]
*tl;dr* `python setup.py --help-commands` shows _Extra commands_ in random order. Only in Python 3.5.

## Problem

Running a local `setup.py` with `--help-commands` parameter lists the _Extra commands_ listing in random order.
```
python setup.py --help-commands
...
Extra commands:
  bdist_wheel       create a wheel distribution
  bdist_egg         create an "egg" distribution
  develop           install package in 'development mode'
  rotate            delete older distributions, keeping N newest files
  alias             define a shortcut to invoke one or more commands
  dist_info         create a .dist-info directory
  install_egg_info  Install an .egg-info directory for the package
  egg_info          create a distribution's .egg-info directory
  test              run unit tests after in-place build (deprecated)
  easy_install      Find/get/install Python packages
  setopt            set an option in setup.cfg or another config file
  upload_docs       Upload documentation to PyPI
  saveopts          save supplied options to setup.cfg or other config file
...
```
Run the same command a second time:
```
Extra commands:
  egg_info          create a distribution's .egg-info directory
  easy_install      Find/get/install Python packages
  install_egg_info  Install an .egg-info directory for the package
  dist_info         create a .dist-info directory
  develop           install package in 'development mode'
  rotate            delete older distributions, keeping N newest files
  upload_docs       Upload documentation to PyPI
  bdist_egg         create an "egg" distribution
  test              run unit tests after in-place build (deprecated)
  setopt            set an option in setup.cfg or another config file
  alias             define a shortcut to invoke one or more commands
  saveopts          save supplied options to setup.cfg or other config file
  bdist_wheel       create a wheel distribution
```

File `setup.py` is bare minimum:
```Python
from setuptools import setup
setup(
    name='MyProgram',
    version='0.1',
    packages=['MyProgram'],
)
```

## Platforms

I tested this on various platforms.

 | OS | `python --version` | `setuptools` version | Issue occurred? |
 | - | - | - | - |
 | Debian 9 | 3.5.3 | 42.0.2 | yes |
 | CentOS 7 | 3.6.8 | 42.0.2 | no |
 | Ubuntu 18 | 3.6.9 | 42.0.2 | no |
 | Windows 10 | 3.8.0 | 42.0.2 | no |

<br />

## Possible Causes

My off-the-cuff guess is there is iterated dictionary somewhere.

> Changed in version 3.7: Dictionary order is guaranteed to be insertion order. This behavior was an implementation detail of CPython from 3.6. <sup>[1](https://docs.python.org/3/library/stdtypes.html#dict)</sup>

**Comment by jaraco** [comment]
Setup.py invocations are largely deprecated and Python 3.7 is the minimum supported version, so this issue is double-obsolete. Thanks for the report and understanding.
